docs/csharp/programming-guide/generics/constraints-on-type-parameters.md

```diff
@@ -10,72 +10,111 @@ helpviewer_keywords:
+|`where T : new()`|The type argument must have a public parameterless constructor. When used together with other constraints, the `new()` constraint must be specified last.|
+|`where T :` *\<base class name>*|The type argument must be or derive from the specified base class.|
+|`where T :` *\<interface name>*|The type argument must be or implement the specified interface. Multiple interface constraints can be specified. The constraining interface can also be generic.|
+|`where T : U`|The type argument supplied for T must be or derive from the argument supplied for U.|
+
+Some of the constraints are mutually exclusive. All value types must have an accessible parameterless constructor. Therefore, the `struct` constraint implies the `new()` constraint and the `new()` constraint cannot be combined with the `struct` constraint. The `unmanaged` contraint implies the `struct` constraint. Therefore, the `unmanaged` constraint cannot be combined with either the `struct` or `new()` constraints.
+
+## Why Use Constraints
+
+If you want to examine an item in a generic list to determine whether it is valid or to compare it to some other item, the compiler must have some guarantee that the operator or method it has to call will be supported by any type argument that might be specified by client code. This guarantee is obtained by applying one or more constraints to your generic class definition. For example, the base class constraint tells the compiler that only objects of this type or derived from this type will be used as type arguments. Once the compiler has this guarantee, it can allow methods of that type to be called in the generic class. Constraints are applied by using the contextual keyword `where`. The following code example demonstrates the functionality we can add to the `GenericList<T>` class (in [Introduction to Generics](introduction-to-generics.md)) by applying a base class constraint.
+
+[!code-csharp[using the class and struct constraints](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#9)]
+
+The constraint enables the generic class to use the `Employee.Name` property because all items of type `T` are guaranteed to be either an `Employee` object or an object that inherits from `Employee`.
+
+Multiple constraints can be applied to the same type parameter, and the constraints themselves can be generic types, as follows:
+
+[!code-csharp[using the class and struct constraints](../../../../samples/snippets/csharp/keywords/GenericWhereConstraints.cs#10)]
+
+By constraining the type parameter, you increase the number of allowable operations and method calls to those supported by the constraining type and all types in its inheritance hierarchy. Therefore, when you design generic classes or methods, if you will be performing any operation on the generic members beyond simple assignment or calling any methods not supported by `System.Object`, you will have to apply constraints to the type parameter.
```

Comment:
Remark: this paragraph fits better to be the first one in the **Why Use Constraints** section